src/lib.rs

```diff
@@ -255,6 +259,11 @@ impl Menu {
         self.0.show_context_menu_for_hwnd(hwnd, x, y)
     }
 
+    #[cfg(target_os = "macos")]
+    pub fn show_context_menu_for_nsview(&self, view: cocoa::base::id) {
```

Comment:
This method should also take an `x`, `y` position as to where to show the menu, does macOS not have this?

Reply:
It's possible to open a context menu at an `x` and `y`, but it's not really needed and it's also quite a bit more complicated because coordinate system conversions will be needed that I haven't quite figured out.

At the moment I'm just opening the menu at the location of the "current event", which should always be where the click was that triggered the menu to be shown.

On the other hand I can see that opening a menu at specified coordinates would be useful for other purposes than just context menus (eg dropdown menus) so I'll come back to that and try and figure out the coordinate system transforms.

Reply:
yeah we need to support x and y, and maybe in the future add a variant that opens under the cursor position

Reply:
Yeah ok, I'll have another look at that.
Do you have any ideas about the coordinate transforms we need? There seems to be a scaling issue (I guess logical vs physical pixels?) as well as flipping the `y` coordinate.

Reply:
no need for conversions, just assume that x and y are logical and relative to the view top-left corner

Reply:
Just providing x and y as coordinates to https://developer.apple.com/documentation/appkit/nsmenu/1518212-popupmenupositioningitem?language=objc doesn't work at all, there's at least a scaling issue for both axes and a sign flip for the y axis (AppKit uses coordinates relative to bottom left 🙄), and possibly also an offset of some sort. 

I'll play with it again later but it was driving me crazy trying to figure it out when I looked at it a few days ago...

Reply:
sign flip should be fine, scaling should also be possible, for instance Windows has `ClientToScreen` api

Reply:
maybe this could help https://developer.apple.com/documentation/uikit/uiview/1622498-convertrect

Reply:
Yeah I did try a few similar methods. I think I'll have to look into how Tao / Winit / whatever crate extracts the coords from the original NSEvent, and do the reverse.

I'll get to it later or tomorrow; for now I'm refactoring a lot of other stuff to make it easier to implement the remaining features.

Reply:
You can leave the context menu for some other time, maybe in a different PR if needed

Reply:
Yes okay, if I don't find an easy way I'll do that. The current method works for context menus anyway, so it's only for popup menus of other kinds that it will be needed.